static/js/narrow.js

```diff
@@ -581,6 +581,10 @@ export function activate(raw_operators, opts) {
     typing_events.render_notifications_for_narrow();
     message_view_header.initialize();
 
+    if (message_lists.current.empty()) {
+        narrow_banner.show_empty_narrow_message();
+    }
```

Comment:
Hmm, I'd expect this to result in the narrow banner appearing temporarily while we're waiting for the server to respond in the case that we don't have any messages in our local cache. This may be a bit tricky to test locally, since usually one has everything cached in the browser, but if you set your browser to a fake 3G network via the Network tab of the Chrome inspector, and do something where `maybe_add_local_messages` will not do anything (say, keyword search?) then I think you'll see this appear immediately and then disappear once the server returns the matching messages.

Reply:
Thanks for this feedback. I was able to test as you outlined above and confirmed the changes I made did in fact load the empty narrow banner immediately and then disappear.